lib/Basics/files.cpp

```diff
@@ -2107,9 +2111,14 @@ int TRI_Crc32File(char const* path, uint32_t* crc) {
   TRI_Free(buffer);
 
   res2 = fclose(fin);
-  if (res2 != TRI_ERROR_NO_ERROR && res2 != EOF) {
+  // According to fclose(3):
+  // > Upon  successful  completion,  0  is returned.  Otherwise, EOF is
+  // > returned and errno is set to indicate the error.
+  // That seems to indicate the condition can never be true. I don't want to
+  // delete it in this PR, but I think it should just be removed.
+  if (res2 != 0 && res2 != EOF) {
     if (res == TRI_ERROR_NO_ERROR) {
-      res = res2;
+      res = ErrorCode{res2};
```

Comment:
I think this would be wrong, even though the code will possibly be never reached.
Here we would be storing whatever fclose's return value was an translate it into one of our own error codes. 
I think instead of all this, we need to query errno if fclose does not return 0, and then
```
res = TRI_set_errno(TRI_ERROR_SYS_ERROR);
```

Reply:
fixed